pints/_abc/_abc_smc.py

```diff
@@ -0,0 +1,176 @@
+#
+# ABC SMC method
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import pints
+import numpy as np
+
+
+class ABCSMC(pints.ABCSampler):
+    """
+    ABC-SMC Algorithm  See, for example, [1]_. In each iteration of the
+    algorithm, the following steps occur::
+
+        theta* ~ p_(t-1)(theta), i.e. sample parameters from previous
+            intermediate distribution
+        theta** ~ K(theta|theta*), i.e. perturb theta* to obtain to new point
+        x ~ p(x|theta**), i.e. sample data from sampling distribution
+        if s(x) < threshold_(t), theta* added to list of samples[t]
+
+    After we have obtained nr_samples samples, t is advanced, and weights
+    are calculated for samples[t-1]. At the last value for threshold,
+    samples are returned whenever they are accepted. This algorithm is
+    also referred to as ABC Population Monte Carlo (ABC PMC) [2].
+
+    References
+    ----------
+    .. [1] "Toni, Tina, et al. Approximate Bayesian computation scheme
+            for parameter inference and model selection in dynamical systems.
+            Journal of the Royal Society Interface, 6.31: 187-202, 2009.
+
+    .. [2] "Beaumont, Mark A., et al. Adaptive approximate Bayesian
+            computation. Biometrika, 96.4: 983-990, 2009."
+    """
+
+    def __init__(self, log_prior, perturbation_kernel=None):
+
+        self._log_prior = log_prior
+        self._samples = [[]]
+        self._accepted_count = 0
+        self._weights = []
+        self._threshold = 1
+        self._e_schedule = [1]
+        self._nr_samples = 100
+        self._xs = None
+        self._ready_for_tell = False
+        self._t = 0
+
+        dim = log_prior.n_parameters()
+
+        if perturbation_kernel is None:
+            self._perturbation_kernel = pints.MultivariateGaussianLogPrior(
+                np.zeros(dim),
+                0.001 * np.identity(dim))
+        elif isinstance(perturbation_kernel, pints.LogPrior):
+            self._perturbation_kernel = perturbation_kernel
+        else:
+            raise ValueError("Provided perturbation kernel must be an instance\
```

Comment:
Not sure how this renders. WIll it have lots of spaces? Usually, you'd break a string like this:
```
        raise ValueError(
            'Provided perturbation kernel must be an instance of'
            ' pints.LogPrior')
```
or
```
        raise ValueError('Provided perturbation kernel must be an instance of'
                         ' pints.LogPrior')
```

Reply:
Note also that (except for docstrings) we use ' instead of " wherever possible (saves us all a bit of shift-key rsi)